### PR TITLE
feat: add base foundation

### PR DESCRIPTION
**Pull request by Skillz4Killz**
# DONT MERGE UNLESS THE FOUNDATION DESIGN IS REVIEWED AND APPROVED!

This pr creates the base foundation for everything.

- Updates the deps to latest.
- Add a bot folder
  - Add a HTTP server handler and necessary configs
  - Add a ready event handler
- Add a gateway folder
- Add a rest folder
- Add a microservices folder
  - Add a analytics folder

### CODE REVIEW

**Comment by nsylke** [comment]
Could we go with environment files instead of config files? Possibly use [dotenv](https://deno.land/x/dotenv)?

**Comment by Skillz4Killz** [comment]
@nsylke Is there any reason to prefer dotenv? It not a big deal but i always thought a .ts file will just inherintly have a better ts experience

**Comment by nsylke** [comment]
@Skillz4Killz More along the lines of following what already exists in current version of TB. However, I am open to change, let me think about it a bit more and we can decide then. As of right now, feel free to leave it as `config.ts`.

File: bot/events/ready.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  if (!application) return;
+
+  // ADDS ALL TEAM MEMBERS AS OWNERS
+  if (application.team)
+    application.team.members.forEach((member) => {
+      botCache.ownerIDs.add(member.user.id!);
+    });
+
+  // ADDS THE OWNER THEMSELF AS OWNER. THE SET WILL REMOVE IF REDUNDANT FROM TEAM.MEMBERS
+  botCache.ownerIDs.add(application.owner.id!);
```
Comment (nsylke) [inline bot/events/ready.ts:22]: Shouldn't this be an if/else statement? If application.team then do X else do Y.

Comment (Skillz4Killz) [reply bot/events/ready.ts:22]: Technically, yes you are right but in essence this is the same thing but imo this is cleaner without if else. I prefer avoiding nested code whenever possible. Performance wise this is basically nill as well as sets/maps are insanely fasst 90/mil ops/s and this is 1 extra op. This isn't worth nesting imo.

If you still prefer if/else let me know and i can change

**Comment by nsylke** [comment]
closing in favor of #5

**Pull request closed without merging**